slice/IceGrid/Admin.ice

```diff
@@ -625,160 +619,143 @@ module IceGrid
         void objectUpdated(ObjectInfo info);
 
-        /// <code>objectRemoved</code> is called to notify an observer when an object registered with the {@link Admin}
-        /// interface was removed.
+       /// Notifies the observer that a well-known object was removed.
         /// @param id The identity of the removed object.
         void objectRemoved(Ice::Identity id);
     }
 
-    /// Used by administrative clients to view, update, and receive observer updates from the IceGrid registry. Admin
-    /// sessions are created either via the {@link Registry} object or via the registry admin
-    /// <code>SessionManager</code> object.
+    /// Represents an administrative session between an admin tool and an IceGrid registry.
     /// @see Registry
     interface AdminSession extends Glacier2::Session
     {
-        /// Keep the session alive.
-        /// As of Ice 3.8, there is no need to call this operation, and its implementation does nothing.
+        /// Keeps the session alive.
+        /// @deprecated As of Ice 3.8, there is no need to call this operation, and its implementation does nothing.
         ["deprecated"]
         idempotent void keepAlive();
 
-        /// Get the admin interface. The admin object returned by this operation can only be accessed by the session.
-        /// @return The admin interface proxy. The returned proxy is never null.
+        /// Gets a proxy to the IceGrid admin object. The admin object returned by this operation can only be accessed
+        /// by the session.
+        /// @return A proxy to the IceGrid admin object. This proxy is never null.
         ["cpp:const"] idempotent Admin* getAdmin();
 
-        /// Get a "template" proxy for admin callback objects. An Admin client uses this proxy to set the category of
+        /// Gets a "template" proxy for admin callback objects. An Admin client uses this proxy to set the category of
         /// its callback objects, and the published endpoints of the object adapter hosting the admin callback objects.
         /// @return A template proxy. The returned proxy is null when the Admin session was established using Glacier2.
         ["cpp:const"] idempotent Object* getAdminCallbackTemplate();
 
-        /// Set the observer proxies that receive notifications when the state of the registry or nodes changes.
+        /// Sets the observer proxies that receive notifications when the state of the registry or nodes changes.
         /// @param registryObs The registry observer.
         /// @param nodeObs The node observer.
         /// @param appObs The application observer.
         /// @param adptObs The adapter observer.
         /// @param objObs The object observer.
-        /// @throws ObserverAlreadyRegisteredException Raised if an observer is already registered with this registry.
+        /// @throws ObserverAlreadyRegisteredException Thrown when an observer is already registered with this registry.
```

Comment:
The idempotent is also arguably correct here - multiple calls with the same args won't change the object state. However, the client can get a very confusing ObserverAlreadyRegisteredException when such a retry occurs.